tests/test_rocket.py

```diff
@@ -76,3 +76,89 @@ def mainTrigger(p, y):
     )
 
     assert test_rocket.allInfo() == None
+
+
+@patch("matplotlib.pyplot.show")
+def test_airfoil(mock_show):
+    test_motor = SolidMotor(
+        thrustSource="data/motors/Cesaroni_M1670.eng",
+        burnOut=3.9,
+        grainNumber=5,
+        grainSeparation=5 / 1000,
+        grainDensity=1815,
+        grainOuterRadius=33 / 1000,
+        grainInitialInnerRadius=15 / 1000,
+        grainInitialHeight=120 / 1000,
+        nozzleRadius=33 / 1000,
+        throatRadius=11 / 1000,
+        interpolationMethod="linear",
+    )
+
+    test_rocket = Rocket(
+        motor=test_motor,
+        radius=127 / 2000,
+        mass=19.197 - 2.956,
+        inertiaI=6.60,
+        inertiaZ=0.0351,
+        distanceRocketNozzle=-1.255,
+        distanceRocketPropellant=-0.85704,
+        powerOffDrag="data/calisto/powerOffDragCurve.csv",
+        powerOnDrag="data/calisto/powerOnDragCurve.csv",
+    )
+
+    test_rocket.setRailButtons([0.2, -0.5])
+
+    NoseCone = test_rocket.addNose(
+        length=0.55829, kind="vonKarman", distanceToCM=0.71971
+    )
+    FinSetNACA = test_rocket.addFins(
+        2,
+        span=0.100,
+        rootChord=0.120,
+        tipChord=0.040,
+        distanceToCM=-1.04956,
+        airfoil=("tests/fixtures/airfoils/NACA0012-radians.txt", "radians"),
+    )
+    FinSetE473 = test_rocket.addFins(
+        2,
+        span=0.100,
+        rootChord=0.120,
+        tipChord=0.040,
+        distanceToCM=-1.04956,
+        airfoil=("tests/fixtures/airfoils/e473-10e6-degrees.csv", "degrees"),
+    )
+    Tail = test_rocket.addTail(
+        topRadius=0.0635, bottomRadius=0.0435, length=0.060, distanceToCM=-1.194656
+    )
+
+    def drogueTrigger(p, y):
+        # p = pressure
+        # y = [x, y, z, vx, vy, vz, e0, e1, e2, e3, w1, w2, w3]
+        # activate drogue when vz < 0 m/s.
+        return True if y[5] < 0 else False
+
+    def mainTrigger(p, y):
+        # p = pressure
+        # y = [x, y, z, vx, vy, vz, e0, e1, e2, e3, w1, w2, w3]
+        # activate main when vz < 0 m/s and z < 800 m.
+        return True if y[5] < 0 and y[2] < 800 else False
+
+    Main = test_rocket.addParachute(
+        "Main",
+        CdS=10.0,
+        trigger=mainTrigger,
+        samplingRate=105,
+        lag=1.5,
+        noise=(0, 8.3, 0.5),
+    )
+
+    Drogue = test_rocket.addParachute(
+        "Drogue",
+        CdS=1.0,
+        trigger=drogueTrigger,
+        samplingRate=105,
+        lag=1.5,
+        noise=(0, 8.3, 0.5),
+    )
+
+    assert test_rocket.allInfo() == None
```

Comment:
I believe it would be quite important to assert the static margin, for both cases, with and without airfoils. What do you think?

Reply:
I added an `assert static_magin<2` in both the tests in the rocket_tests.py which I think is fair but maybe there is a better way to do it. Any suggestions?

Reply:
Aah that is smart. I changed the interval to 0.01, I think it is more reasonable